### PR TITLE
Add ArmClientException to SDK

### PR DESCRIPTION
**Pull request by melutzky**
this is a fix of https://mseng.visualstudio.com/AppInsights/_workitems/edit/1313694

### CODE REVIEW

File: src/SDK/SmartDetectorsSDK/Arm/ArmClientException.cs [hunk 1]
```diff
@@ -1,68 +1,68 @@
...
     using System.Net;
     using System.Runtime.Serialization;
 
     /// <summary>
-    /// Exception thrown whenever an error is encountered when calling the <see cref="IAzureResourceManagerClient"/> operations.
+    /// Exception thrown whenever an error is encountered when calling the <see cref="IArmClient"/> operations.
     /// </summary>
     [Serializable]
-    public class AzureResourceManagerClientException : Exception
+    public class ArmClientException : Exception
```
Comment (EranGon) [inline src/SDK/SmartDetectorsSDK/Arm/ArmClientException.cs]: >ArmClientException [](start = 17, length = 18)

please rename this back - AzureResourceManagerClientException is a much better name

while you're at it, also change IArmClient to IAzureResoruceManagerClient

and call me when you do it, so I'll show you the short way to do it

File: src/analysis/SmartDetectorsAnalysis/SmartDetectorRunner.cs [hunk 2]
```diff
@@ -46,15 +46,15 @@ public class SmartDetectorRunner : ISmartDetectorRunner
         /// <param name="smartDetectorRepository">The Smart Detector repository</param>
         /// <param name="smartDetectorLoader">The Smart Detector loader</param>
         /// <param name="analysisServicesFactory">The analysis services factory</param>
-        /// <param name="azureResourceManagerClient">The azure resource manager client</param>
+        /// <param name="azureResourceManagerClient">The Azure resource manager client</param>
```
Comment (EranGon) [inline src/analysis/SmartDetectorsAnalysis/SmartDetectorRunner.cs]: >resource manager [](start = 63, length = 16)

capitalize this while you're at it

File: src/SDK/SmartDetectorsShared/Clients/ActivityLogClient.cs [hunk 3]
```diff
@@ -30,21 +30,21 @@ public class ActivityLogClient : IActivityLogClient
...
         private readonly IExtendedTracer tracer;
         private readonly Policy retryPolicy;
-        private readonly IArmClient armClient;
+        private readonly IAzureResourceManagerClient azureResourceManagerClient;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ActivityLogClient"/> class.
         /// </summary>
-        /// <param name="armClient">The ARM client</param>
+        /// <param name="azureResourceManagerClient">The Azure resource manager client</param>
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/ActivityLogClient.cs]: >resource manager [](start = 63, length = 16)

capitalize

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 4]
```diff
@@ -37,9 +37,9 @@ public class AnalysisServicesFactory : IInternalAnalysisServicesFactory
         /// <param name="tracer">The tracer</param>
         /// <param name="httpClientWrapper">The HTTP client wrapper.</param>
         /// <param name="credentialsFactory">The credentials factory.</param>
-        /// <param name="azureResourceManagerClient">The azure resource manager client.</param>
+        /// <param name="azureResourceManagerClient">The Azure resource manager client.</param>
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: >resource manager [](start = 63, length = 16)

capitalize

File: src/SDK/SmartDetectorsShared/Clients/IAzureResourceManagerClientInternal.cs [hunk 5]
```diff
@@ -13,9 +13,9 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.Clients
     using Microsoft.Azure.Monitoring.SmartDetectors.Arm;
 
     /// <summary>
-    /// An interface for azure resource manager client
+    /// An interface for Azure resource manager client
     /// </summary>
-    public interface IAzureResourceManagerClient : IArmClient
+    public interface IAzureResourceManagerClientInternal : IAzureResourceManagerClient
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/IAzureResourceManagerClientInternal.cs]: >IAzureResourceManagerClientInternal [](start = 21, length = 35)

Sorry for making your do this again, but the internal suffix is too ugly
Please rename this interface to IExtendedAzureResourceManagerClient and the implementing class
ExtendedAzureResourceManagerClient

**Comment by melutzky** [comment]
Created new one

**Pull request closed without merging**